### PR TITLE
(#4668) Add simd flags to fftw.

### PR DESCRIPTION
**Pull request by ruilvo**
Specify library name and version:  **fftw/3.3.9**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ruilvo** [comment]
Fixes #4668 and conan-io/cmake-conan#311.

File: recipes/fftw/all/conanfile.py
Comment (SpaceIm) [inline recipes/fftw/all/conanfile.py:30]:
```suggestion
                       "SIMD": "auto"}
```

No auto as default value of such option please. It may build binaries that consumers can't use on their arch.

File: recipes/fftw/all/patches/fix-simd.patch
Comment (SpaceIm) [inline recipes/fftw/all/patches/fix-simd.patch:7]: Could you explain this patch? What does it change in the context of conan? Seems useless, but maybe I'm wrong.

File: recipes/fftw/all/conanfile.py
Comment (SpaceIm) [inline recipes/fftw/all/conanfile.py:23]: prefer lowercase values please

File: recipes/fftw/all/conanfile.py
Comment (ruilvo) [reply recipes/fftw/all/conanfile.py:30]: "auto" does nothing. I did it on purpose to exactly not break existing compilations. `auto` in fact does nothing. Do you have a better suggestion for a name?

File: recipes/fftw/all/patches/fix-simd.patch
Comment (ruilvo) [reply recipes/fftw/all/patches/fix-simd.patch:7]: The patch doesn't fix anything in terms of conan itself, but it does fix the part of the build I'm trying to enable with this PR. At least on my machine, I have to unset the variable both with `CACHE` and without or otherwise it doesn't test the second flag, and so fails to do anything on Windows.

File: recipes/fftw/all/conanfile.py
Comment (ruilvo) [reply recipes/fftw/all/conanfile.py:23]: Will do.

File: recipes/fftw/all/conanfile.py
Comment (ruilvo) [reply recipes/fftw/all/conanfile.py]: I went with your suggestion of using False instead.

File: recipes/fftw/all/patches/fix-simd.patch
Comment (SpaceIm) [reply recipes/fftw/all/patches/fix-simd.patch:7]: It's weird, is there an issue opened upstream?

I'll test it on my machine when I have time.

Comment (ruilvo) [reply recipes/fftw/all/patches/fix-simd.patch:7]: It is now. I opened it myself after detecting the problem while making this PR.

The minimum you need to test it on a CMakeLists.txt:
```cmake
include (CheckCCompilerFlag)
foreach (FLAG "-msse2" "/arch:SSE2")
  unset (HAVE_SSE2 CACHE)
  #unset(HAVE_SSE2)
  check_c_compiler_flag (${FLAG} HAVE_SSE2)
  if (HAVE_SSE2)
    set (SSE2_FLAG ${FLAG})
    break()
  endif ()
endforeach ()
```
And you'll notice it only tests once (the first flag), then fails, of course, because it didn't test for the MSVC version of the flag. At least on my machine it does it. Only unsetting twice it works.

Comment (ruilvo) [reply recipes/fftw/all/patches/fix-simd.patch:7]: You can probably reproduce it on a *NIX machine just by inverting the order of the flags, so it fails first.

Comment (ruilvo) [reply recipes/fftw/all/patches/fix-simd.patch:7]: They merged [my PR](https://github.com/FFTW/fftw3/pull/232) upstream, so the patch should be a fix just up to the next version.

Comment (uilianries) [reply recipes/fftw/all/patches/fix-simd.patch:7]: As it is a PR related to build and accepted by the upstream, it can be accepted.

**Review by uilianries — APPROVED** [review]
Thank you for you contribution. This PR should be merged as soon as we have all approvals and your profile added to EAP (next monday).

File: recipes/fftw/all/patches/fix-simd.patch
Comment (SpaceIm) [reply recipes/fftw/all/patches/fix-simd.patch:7]: :+1:

**Review by SpaceIm — DISMISSED** [review]

File: recipes/fftw/all/conanfile.py
Comment (SpaceIm) [reply recipes/fftw/all/conanfile.py]: Oh ! I forgot: this patch should also be propagated to 3.3.8

File: recipes/fftw/all/patches/fix-simd.patch
Comment (mathbunnyru) [reply recipes/fftw/all/patches/fix-simd.patch:7]: May I ask you to add the PR in the header of the patch?

**Review by mathbunnyru — APPROVED** [review]

File: recipes/fftw/all/conandata.yml
Comment (mathbunnyru) [inline recipes/fftw/all/conandata.yml:16]: Let's fix 3.3.8 as well, otherwise we will have options which are present in 3.3.8, but do not work in some cases.

**Review by mathbunnyru — CHANGES_REQUESTED** [review]

File: recipes/fftw/all/conanfile.py
Comment (ruilvo) [reply recipes/fftw/all/conanfile.py]: @SpaceIm changes have been pushed to 3.3.8 as well.

File: recipes/fftw/all/conandata.yml
Comment (ruilvo) [reply recipes/fftw/all/conandata.yml:16]: @mathbunnyru changes have been applied to 3.3.8. What do you mean with "May I ask you to add the PR in the header of the patch?", sorry?

File: recipes/fftw/all/patches/fix-simd.patch
Comment (mathbunnyru) [inline recipes/fftw/all/patches/fix-simd.patch:1]:
```suggestion
https://github.com/FFTW/fftw3/pull/232
--- a/CMakeLists.txt
```

**Review by mathbunnyru — CHANGES_REQUESTED** [review]
@ruilvo I've seen it several times in patches, that the upstream commit or PR is added to make it easier to see, that it was already approved / submitted to the upstream.

I made a small fix suggestion to make it easy for you.

File: recipes/fftw/all/conandata.yml
Comment (mathbunnyru) [reply recipes/fftw/all/conandata.yml:16]: Explained below.

File: recipes/fftw/all/patches/fix-simd.patch
Comment (ruilvo) [reply recipes/fftw/all/patches/fix-simd.patch:2]: Done, thanks!

**Review by mathbunnyru — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by ruilvo** [comment]
So, now I'm on the EAP, how can I trigger the CI to build?

**Comment by SSE4** [comment]
> So, now I'm on the EAP, how can I trigger the CI to build?

- close PR, wait 10 seconds, and re-open
- do an empty commit (`git commit --allow-empty`)
- do force push (`git commit --amend && git push -f`)

**Comment by ruilvo** [comment]
Uh... Question, now that it's passing, how do I get this merged?

**Review by SpaceIm — APPROVED** [review]

**Comment by SpaceIm** [comment]
It should be merged quickly since there are 3 approvals.

**Review by SSE4 — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Comment by SpaceIm** [comment]
@SSE4 @uilianries @danimtb this PR seems to require manual merge.

**Comment by ericLemanissier** [comment]
This is github being confused by empty commits. There was only 2 reviews after https://github.com/conan-io/conan-center-index/pull/4671/commits/490a4b62d5fe18f33418edd25a8c125a6a5fec1c

**Review by mathbunnyru — APPROVED** [review]

**Comment by mathbunnyru** [comment]
Approved once again just in case.
Empty commits are evil.

**Comment by ruilvo** [comment]
Seems that I screwed up with that empty commit... I'll do the "close/open" technique the next time. Sorry!

**Comment by danimtb** [comment]
@conan-center-bot 💌

**Comment by uilianries** [comment]
empty commit vs Jenkins

**Pull request merged**
